numpy/core/src/multiarray/experimental_public_dtype_api.c

```diff
@@ -180,11 +180,78 @@ PyArrayInitDTypeMeta_FromSpec(
          * It is up to the user to get this right, and slots are sorted
          * exactly like they are stored right now:
          */
-        void **current = (void **)(&(
-                NPY_DT_SLOTS(DType)->discover_descr_from_pyobject));
-        current += slot - 1;
-        *current = pfunc;
+        if (slot <= 8) {
```

Comment:
i.e. this 8 is the `_NPY_NUM_DTYPE_SLOTS`  any greater ones may be arrfuncs.  If we don't use `_NPY_NUM_DTYPE_SLOTS` the define could just go (but I prefer it, easier to remember to update it there.)